x/provider/types/provider_test.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+import (
+	"fmt"
+	hubtypes "github.com/sentinel-official/hub/types"
+	"testing"
+)
+
+func TestProvider_Validate(t *testing.T) {
+	correctAddress := hubtypes.ProvAddress{}
+
+	tests := []struct {
+		name string
+		p    Provider
+		want error
+	}{
+		{"address zero", Provider{"", "", "", "", ""}, errInvalidProvName},
+		{"address empty", Provider{correctAddress.String(), "", "", "", ""}, errInvalidProvName},
+		{"name length zero", Provider{correctAddress.String(), "", "", "", ""}, errInvalidProvName},
+		{"name length greater than 64", Provider{correctAddress.String(), GT64, "", "", ""}, errInvalidProvName},
+		{"identity length greater than 64", Provider{correctAddress.String(), "name", GT64, "", ""}, errInvalidProvIdentity},
+		{"website length greater than 64", Provider{correctAddress.String(), "name", "", GT64, ""}, errInvalidProvWebsite},
+		{"description length greater than 256", Provider{correctAddress.String(), "name", "", "", GT256}, errInvalidProvDescription},
+		{"valid", Provider{correctAddress.String(), "name", "", "", ""}, nil},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.p.Validate(); err != nil && err.Error() != tt.want.Error() {
+				t.Errorf("Validate() error = %v", err)
+			}
+		})
+	}
+}
+
+var (
```

Comment:
@bsrinivas8687 is it okay to keep these here? or should we create them as a shared var and use it in the `Validate()` method and in this test case?

Reply:
You can just check whether error is nil or not. Need not to check error message